src/Common/tests/System/PlatformDetection.cs

```diff
@@ -17,7 +17,9 @@ public static class PlatformDetection
         public static bool IsOSX { get; } = RuntimeInformation.IsOSPlatform(OSPlatform.OSX);
         public static bool IsNetBSD { get; } = RuntimeInformation.IsOSPlatform(OSPlatform.Create("NETBSD"));
         public static bool IsNotWindowsNanoServer { get; } = (IsWindows &&
-                File.Exists(Path.Combine(Environment.GetEnvironmentVariable("windir"), "regedit.exe")));
+            File.Exists(Path.Combine(Environment.GetEnvironmentVariable("windir"), "regedit.exe")));
+        public static bool IsWindows10Version1607 { get; } = IsWindows &&
+            GetWindowsVersion() == 10 && GetWindowsMinorVersion() == 0 && GetWindowsBuildNumber() == 14393;
```

Comment:
Should this be an "or better" check?  Don't want the tests to hurt people on the fast train by having too large of a build number.